docs/patterns/components/TwoColumnPanel/index.js

```diff
@@ -0,0 +1,361 @@
+
+      <h2>Basic</h2>
+
+      <TwoColumnPanel>
+        <ColumnLeft aria-labelledby="sidebar">
+          <ColumnHeader id="sidebar">Items</ColumnHeader>
+          <nav aria-label="Sidebar navigation">
+            <ul>
+              {items.map(item => (
+                <li key={item.id}>
+                  <a
+                    href="#"
+                    onClick={selectItem(item.id)}
+                    aria-current={item.id === selectedItem.id}
+                  >
+                    {item.name}
+                    <em>{item.description}</em>
+                  </a>
+                </li>
+              ))}
+            </ul>
+          </nav>
+        </ColumnLeft>
+        <ColumnRight aria-labelledby="item-header">
+          <ColumnHeader id="item-header">{selectedItem?.name}</ColumnHeader>
+          <div>
+            {selectedItem?.contents}
+            <p>
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque
+              eu suscipit tellus. Sed ultrices pretium vulputate. Maecenas id
+              diam tortor. Integer a felis dignissim, euismod mauris a,
+              vestibulum leo. Aenean vel fringilla turpis. In magna massa,
+              lacinia quis arcu ut, tincidunt condimentum tortor. Maecenas quis
+              vestibulum libero, eget gravida tortor. Suspendisse sed mollis
+              lorem. Pellentesque nec sagittis risus. Nunc faucibus suscipit
+              libero eget vehicula. Praesent scelerisque vitae risus ut auctor.
+              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+              Pellentesque efficitur sollicitudin ornare.
+            </p>
+          </div>
+        </ColumnRight>
+      </TwoColumnPanel>
+
+      <h3>Example</h3>
+
+      <Code>{`<TwoColumnPanel>
+  <ColumnLeft aria-labelledby="sidebar">
+    <ColumnHeader id="sidebar">Items</ColumnHeader>
+    <nav aria-label="Sidebar navigation">
+      <ul>
+        {items.map(item => (
+          <li key={item.id}>
+            <a
+              href={item.url}
+              onClick={selectItem(item.id)} aria-current={item.id === selectedItem.id}
+              aria-current={item.id === selectedItem.id}
+            >
+              {item.name}
+              <em>{item.description}</em>
+            </a>
+          </li>
+        ))}
+      </ul>
+    </nav>
+  </ColumnLeft>
+  <ColumnRight aria-labelledby="item-header">
+    <ColumnHeader id="item-header">{selectedItem?.name}</ColumnHeader>
+    <div>
+      {selectedItem?.contents}
+    </div>
+  </ColumnRight>
+</TwoColumnPanel>`}</Code>
+
+      <h2>With Optional Group Heading</h2>
+
+      <TwoColumnPanel>
+        <ColumnLeft aria-labelledby="group-heading">
+          <ColumnHeader>Grouped Items</ColumnHeader>
+          <ColumnList>
+            <ColumnGroupHeader id="group-heading">
+              <h3>Optional group heading</h3>
+            </ColumnGroupHeader>
+            <nav aria-label="Sidebar group 1 navigation">
```

Comment:
https://github.com/dequelabs/cauldron/issues/488 mentions supporting vertical tabs. It might be nice to include an example of how to do that in a semantic tab + tab panel way.

Reply:
@schne324 and I went back and forth on this a couple of times and decided not to support tabs because there were some significant accessibility concerns with supporting a tab panel:

* `TabPanel` has an aria relationship with its associated tabs meaning that hiding the tabs potentially has accessibility issues at worst, or confusing at the least when the tabs are hidden
* Depending on how the left panel was hidden we would have a couple of options of ensuring hidden content was not able to be navigated to: We could use `inert` with a polyfill, `tabindex="-1`" on other focusable elements, or implement a focus trap.

We decided to go with a focus trap, and given that it's not a common pattern to have a tab's controls hidden the focus trap + hiding the panel felt like the least worst option. If you think it's worth documenting some of these decisions in the component I'm happy to oblige.

Reply:
Yeah, I think documenting the usage/limitations with reasonings of the component would be good for posterity.